### PR TITLE
FIX Manually read token from .transifexrc

### PR DESCRIPTION
**Pull request by emteknetnz**
There seems to be a bug in the tx binary where it won't read a previously set API token in ~/.transifexrc - there is an existing issue for this https://github.com/transifex/cli/issues/233

This PR works around this limitation by manually reading .transifexrc and passing it in as an environment variable to sub processes

### CODE REVIEW

**Review by GuySartorelli — COMMENTED** [review]
I don't like this at all. Can you please give me steps to reproduce the bug we're working around so I can understand this better?

**Comment by emteknetnz** [comment]
I've got a fresh install of ubuntu so have also got the latest version of go sudo apt install golang-go) and the tx binary (1.6.17)

To create ~/.transifexrc I simply run `tx push -s` (or maybe it was a tx pull command, cannot remember exactly), the binary prompted me for my transifix token which I entered, ~/.transifexrc was created and contain my token

Running `tx push -s` subsequently prompted me for my token again, which it shouldn't do, instead it should simply read it from ~/.transifexrc

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
Seems fine, if unfortunate that it's needed.
Please do this in a central place (e.g. in `$this->exec()` if the command is `tx`) so new code will automatically have the correct prepended env var

**Comment by emteknetnz** [comment]
Updated

File: src/Translator.php [hunk 1]
```diff
@@ -837,4 +846,21 @@ private function recursiveKeySort(array &$arr): void
...
     }
+
+    /**
+     * The tx binary doesn't appear to correctly read from ~/.transifexrc, even though it will correctly
+     * write to it after prompting for it if it's missing. This is a workaround to read the token and
+     * use it a TX_TOKEN environment variable, which the tx binary will use.
+     */
+    private function readTxToken(): void
+    {
+        if (!$this->exec('if [ -f ~/.transifexrc ]; then echo 1; else echo 0; fi')) {
```
Comment (ChloeSartorelli) [inline src/Translator.php]: Please use PHP here rather than using bash.

Sorry I didn't notice this last time.

File: src/Translator.php [hunk 2]
```diff
@@ -837,4 +846,21 @@ private function recursiveKeySort(array &$arr): void
...
+            throw new RuntimeException('Could not find ~/.transifexrc');
+        }
+        $contents = $this->exec('cat ~/.transifexrc');
```
Comment (ChloeSartorelli) [inline src/Translator.php]: Please use PHP here rather than bash.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/Translator.php (hunk 1 above)
Comment (emteknetnz) [reply src/Translator.php]: Need to use bash as we're using `~` and we need it to resolve to the user home path

File: src/Translator.php (hunk 2 above)
Comment (emteknetnz) [reply src/Translator.php]: Same as above

File: src/Translator.php (hunk 1 above)
Comment (ChloeSartorelli) [reply src/Translator.php]: You can use `$_SERVER['HOME']` or `getenv('HOME')` or add `symfony/filesystem` as a dependency and use `Path::canonicalize()` (the latter of which would be my preference as it's most robust)

Comment (emteknetnz) [reply src/Translator.php]: Updated

File: src/Translator.php (hunk 2 above)
Comment (emteknetnz) [reply src/Translator.php]: Updated

File: composer.json [hunk 3]
```diff
@@ -17,6 +17,7 @@
         "symfony/process": "^4 || ^5 || ^6",
         "symfony/yaml": "^4 || ^5 || ^6",
         "symfony/dotenv": "^5 || ^6",
+        "symfony/filesystem": "^5 || ^6",
```
Comment (emteknetnz) [inline composer.json:20]: ^5 is required for the PHP 7.4 build. Haven't added ^7 as we'd need to update all the other deps and getting this to work for CMS 6 is out of scope

**Review by GuySartorelli — APPROVED** [review]
LGTM

**Pull request merged**